docs/python_docs/python/tutorials/performance/backend/dnnl/dnnl_quantization.md

```diff
@@ -0,0 +1,304 @@
+- Conv2D + Add => even simpler idea than the previous ones - instead of overwriting output memory, results are added to the output memory. In the simplest terms: `out_mem = conv_result` is replaced by `out_mem += conv_result`.
+
+Above examples are presented as atomic ones, but often they can be combined together, thus two patterns that can be fused in above example are:
+- Conv2D + BatchNorm + ReLU
+- Conv2D + BatchNorm + Add + ReLU
+
+After fusing all patterns, computational graph will be changed to the following one:
+![fused_fp32_model](https://github.com/dmlc/web-data/blob/main/mxnet/tutorials/onednn/quantization_2_0/fused_f32.png?raw=true)
+
+
+
+### Operator fusion in MXNet
+Since the version 1.6 of MXNet built with oneDNN support, operator fusion had been enabled by default for executing model with Module API, however in version 2.0 it has been decided to remove setting this feature by environment flag and replace it by aware user API call.
+
+To fuse model in MXNet 2.0 there are two requirements:
+- the model must be defined as a subclass of HybridBlock or Symbol,
+- the model must have specific operator patterns which can be fused.
+
+As an example we define example network (sample block from ResNet architecture):
+
+```
+import mxnet as mx
+from mxnet.gluon import nn
+
+class SampleBlock(nn.HybridBlock):
+    def __init__(self):
+        super(SampleBlock, self).__init__()
+        self.conv1 = nn.Conv2D(channels=64, kernel_size=3, strides=1, padding=1,
+                               use_bias=False, in_channels=64)
+        self.bn1 = nn.BatchNorm()
+        self.conv2 = nn.Conv2D(channels=64, kernel_size=3, strides=1, padding=1,
+                               use_bias=False, in_channels=64)
+        self.bn2 = nn.BatchNorm()
+
+    def forward(self, x):
+        out = mx.npx.activation(self.bn1(self.conv1(x)), 'relu')
+        out = self.bn2(self.conv2(out))
+        out = mx.npx.activation(out + x, 'relu')
+        return out
+        
+net = SampleBlock()
+net.initialize()
+
+data = mx.np.zeros(shape=(1,64,224,224))
+# run fusion
+net.optimize_for(data, backend='ONEDNN')
+
+# We can check fusion by plotting current symbol of our optimized network
+sym, _ = net.export(None)
+graph = mx.viz.plot_network(sym, save_format='jpg')
+graph.view()
+```
+Both HybridBlock and Symbol classes provide API to easily run fusion of operators. Single line of code is enabling fusion passes on model:
+```
+net.optimize_for(data, backend='ONEDNN')
+```
+
+*optimize_for* function is available also as Symbol class method. Example call to this API is shown below. Notice that Symbol’s *optimize_for* method is not done in-place, so assigning it to a new variable is required:
+
+```
+optimized_symbol = sym.optimize_for(backend='ONEDNN')
+```
+
+For the above model definition in a naive benchmark with artificial data, we can gain up to 1.75x speedup without any accuracy loss on our testing machine with Intel(R) Core(TM) i9-9940X.
+
+
+## Quantization
```

Comment:
Inconsistent new line after ## and ### - seems to be not too important as in HackMD preview it does look identical with or without it.

Reply:
Done